sdk/core/core/inc/az_http_client.h

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_HTTP_CLIENT_H
+#define AZ_HTTP_CLIENT_H
+
+#ifdef MOCK_CURL
+#include <az_mock_curl.h>
+#else
+#include <az_curl_adapter.h>
```

Comment:
IMHO, it should be in `.c` file instead of header.

Reply:
@gilbertw @vhvb1989 I'm trying hard to reduce a number of binaries. Each such flag doubles the number so it's growing exponentially. Currently I'm releasing Boost C++ library binaries as a NuGet package and it's not fun because it x86/x64 * thread model * linkage type * stl type *…. It's not fun. If we our policies will be runtime changeable then it will be a peace of cake to replace transport layer. So you don't need another compile time flag.